src/Fantomas.Core/CodePrinter.fs

```diff
@@ -1553,9 +1553,24 @@ let genExpr (e: Expr) =
     | Expr.IndexFromEnd node -> !- "^" +> genExpr node.Expr |> genNode node
     | Expr.Typar node -> genSingleTextNode node
     | Expr.DotLambda node ->
+        let genDotLambdaExpr expr =
+            fun ctx ->
+                let currentConfig = ctx.Config
+
+                let dotLambdaConfig =
+                    { currentConfig with
```

Comment:
Hmm, I'm a little torn about this one.
The thinking is fine, but overriding the config will open the door for the next issue to be reported:

```fsharp
[<Test>]
let ``yow`` () =
    formatSourceString
        """
let Meh () = 1

type Bar() =
    member this.Foo(v:int):int = v + 1

let b = Bar()
b |> _.Foo(Meh ())
"""
        { config with
            SpaceBeforeUppercaseInvocation = true }
    |> prepend newline
    |> should
        equal
        """
let Meh () = 1

type Bar() =
    member this.Foo(v: int) : int = v + 1

let b = Bar ()
b |> _.Foo(Meh ())
"""
```

`expr` in this case is `Meh ()` can will no longer be respected.
Checking if the `node.Expr` is parenthesis would be safer here.

Reply:
Thanks, much better that way.